### DISCUSSION TITLE
alternative architectures?

### DISCUSSION THREAD

**Discussion by guiv42** [helge17/tuxguitar#137] [General]
TuxGuitar release 1.6.0 has been accepted in the official repository of openSUSE Tumbleweed.
After I submitted this package, openSUSE maintainer has added several patches to enable builds on alternative architectures: aarch64, armv6hl, armv7hl, i586, ppc64le. So now, rpm packages of 1.6.0 are available for all these architectures in openSUSE repo (these are dev packages, only the x86_64 is available in main repo).

I imagine that some users might be interested by such builds (e.g. [aarch64](https://sourceforge.net/p/tuxguitar/feature-requests/97/) was requested in SF). Would it make sense to "up-stream" such builds here in this repo?

My main problem is I cannot test them... 
If someone could install OpenSUSE Tumbleweed on a machine with such an architecture, then it should be possible to test the corresponding rpm package. Dev packages are available there: https://software.opensuse.org//download.html?project=Java%3Apackages&package=tuxguitar
Click on "openSUSE", then "grab binary packages directly"

**Comment by helge17** [comment]
On Debian, TuxGuitar 1.5.6 is also available for different architectures, but the approach to support the architectures is different.
OpenSuse adds a build script for each architecture, see https://build.opensuse.org/package/show/Java:packages/tuxguitar .
Debian uses one build script tuxguitar-linux-x86_64 for all architectures:
https://salsa.debian.org/java-team/tuxguitar/-/tree/master/build-scripts
https://salsa.debian.org/java-team/tuxguitar/-/tree/master/debian/patches
Provide packages for different Linux architectures is possible, but quite a lot of work: setting up the build systems (I once set up two Qemu VMs for arm64 and ppc64el on x86_64 hardware), customizing the build process, etc. We already have quite a few TuxGuitar "flavors": Linux, Windows, MacOS with SWT/JFX/QT (and Android). So currently I would let the distributions take care of the non-x86_64 architectures and maybe take care of them later, e.g. after deciding on one of the GUI toolkits SWT/JFX/QT?

**Comment by guiv42** [comment]
OK, clear !
